### ISSUE TITLE
Differential Transformer loss spikes while training.

### ISSUE THREAD

**Issue by fasil-saidalavi** [microsoft/unilm#1718]
I have trained a 1.3B model using both the Differential Transformer and the standard Transformer. I observed a slight improvement in LLM evaluation scores for the Differential Transformer variant, and its loss was consistently lower. However, when I tried the 7B model comparison, training showed loss spikes, and the loss started increasing after a certain point. I also noticed that the gradient norm increased at the same points where the loss spiked. In terms of implementation details, I only used the differential attention part from this repository; I did not include the SwiGLU layer. Instead, I used a standard FFN layer.

**Comment by YTianZHU** [comment]
Hi @fasil-saidalavi , sorry for the late response. Do you use same FFN implementation for both Diff and Transformer? Would you post a code snippet of attention implementation of both Diff and baseline Transformer?

**Comment by fasil-saidalavi** [comment]
`
print("hello world")
`

**Comment by donglixp** [comment]
Diff Attention V2: https://spiky-homegrown-4cb.notion.site/Differential-Transformer-V2-2e7baa052def80ecaa93d4d67d125417
